lib/npm/script/run.js

```diff
@@ -22,7 +22,16 @@ function runScript(context, script, msg, next) {
       return next(err);
     }
     var options = createOptions(context.path, context);
-    var proc = spawn(_path, [], options);
+    var scriptExtension = path.extname(_path);
+    var proc = '';
+    if ((scriptExtension === '.bat' || scriptExtension === '.cmd') && process.platform === 'win32' || scriptExtension === '.sh' && process.platform != 'win32') {
```

Comment:
Maybe `['.bat','.cmd'].indexOf(scriptExtension) !== -1` rather than `(scriptExtension === '.bat' || scriptExtension === '.cmd')`? Not that much better tbh, so up to you.